File: Math/2894. Divisible and Non-divisible Sums Difference/2894. Divisible and Non-divisible Sums Difference.py
class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        num1 = 0
        num2 = 0
        for n in range(1, n + 1):
            if n % m == 0:
                num2 += n
            else:
                num1 += n

        return num1 - num2

# Time: O(n)
# Space: O(1)

File: Math/2894. Divisible and Non-divisible Sums Difference/2894. Divisible and Non-divisible Sums Difference.js
/**
 * @param {number} n
 * @param {number} m
 * @return {number}
 */
var differenceOfSums = function (n, m) {
  let num1 = 0;
  let num2 = 0;
  for (let i = 1; i < n + 1; i++) {
    if (i % m === 0) num2 += i;
    else num1 += i;
  }
  return num1 - num2;
};

// Time: O(n)
// Space: O(1)
